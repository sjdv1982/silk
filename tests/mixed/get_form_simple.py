import sys
import numpy as np
from silk.mixed .get_form import get_form
import json

_print = print
def print(*args):
    for a in args:
        _print(json.dumps(a, indent=2, sort_keys=True), end=" ")
    _print()

storage, form = get_form(1)
print(storage, form); print()

storage, form = get_form("test")
print(storage, form); print()

storage, form = get_form(["test", "test2"])
print(storage, form); print()

storage, form = get_form([1, 10.0])
print(storage, form); print()

storage, form = get_form(["test", 1, True])
print(storage, form); print()

storage, form = get_form({
    "a": None,
    "b": False,
    "c": 10,
    "d": 20.0,
    "e": "OK",
    "f": [1,2],
})
print(storage, form); print()

storage, form = get_form(np.zeros(10, dtype=np.int32))
print(storage, form); print()

storage, form = get_form(np.zeros(20, dtype=np.bool))
print(storage, form); print()

dt = np.dtype([
    ("a", int),
    ("b", float),
], align=True)
storage, form = get_form(np.zeros(10, dtype=dt))
print(storage, form); print()
