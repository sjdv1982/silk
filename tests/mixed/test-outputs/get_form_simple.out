"pure-plain" "integer" 

"pure-plain" "string" 

"pure-plain" {
  "identical": true,
  "items": "string",
  "shape": [
    2
  ],
  "type": "array"
} 

"pure-plain" {
  "identical": false,
  "items": [
    "integer",
    "number"
  ],
  "shape": [
    2
  ],
  "type": "array"
} 

"pure-plain" {
  "identical": false,
  "items": [
    "string",
    "integer",
    "boolean"
  ],
  "shape": [
    3
  ],
  "type": "array"
} 

"pure-plain" {
  "properties": {
    "a": "null",
    "b": "boolean",
    "c": "integer",
    "d": "number",
    "e": "string",
    "f": {
      "identical": true,
      "items": "integer",
      "shape": [
        2
      ],
      "type": "array"
    }
  },
  "type": "object"
} 

"pure-binary" {
  "contiguous": true,
  "identical": true,
  "items": {
    "bytesize": 4,
    "type": "integer",
    "unsigned": false
  },
  "shape": [
    10
  ],
  "type": "array"
} 

"pure-binary" {
  "contiguous": true,
  "identical": true,
  "items": {
    "bytesize": 1,
    "type": "boolean"
  },
  "shape": [
    20
  ],
  "type": "array"
} 

"pure-binary" {
  "contiguous": true,
  "identical": true,
  "items": {
    "bytesize": 16,
    "order": [
      "a",
      "b"
    ],
    "properties": {
      "a": {
        "bytesize": 8,
        "type": "integer",
        "unsigned": false
      },
      "b": {
        "bytesize": 8,
        "type": "number"
      }
    },
    "type": "object"
  },
  "shape": [
    10
  ],
  "type": "array"
} 

